.github/workflows/release.yml

```
@@ -30,6 +30,19 @@ jobs:
           VERSION_NUMBER: ${{ github.event.inputs.version_number }}
           COMMIT_ID: ${{ github.event.inputs.commit_id }}
         run: git checkout -b "$VERSION_NUMBER" "$COMMIT_ID"
+      - name: Update version number in source files
+        env:
+          VERSION_NUMBER: ${{ github.event.inputs.version_number }}
+          REPO_NAME: ${{ github.event.repository.name }}
+          source_folder_list: "source test"
+        run: |
+            echo "${{ env.source_folder_list }}" | \
+            xargs -n 1 sh -c \
+            'find $1 -type f \( -name "*.c" -o -name "*.h" \) \
+            -exec sed -i -b -E "0,/^ \* FreeRTOS\+TCP/s/^ \* FreeRTOS\+TCP.*/ \* FreeRTOS\+TCP $VERSION_NUMBER/g" {} +'
```

Can we run this locally to see whether there are no unforeseen consequences?
We might have comments which start with `FreeRTOS+TCP...` and we might end up replacing them.

Also, I might be wrong here, isn't there supposed to be only one instance of the version number info in the file at the top? If so, then it might be better if we remove the `g` at the end (signifying all instance replacement) in the sed command like so:

```suggestion
            -exec sed -i -b -E "0,/^ \* FreeRTOS\+TCP/s/^ \* FreeRTOS\+TCP.*/ \* FreeRTOS\+TCP $VERSION_NUMBER/" {} +'
```
Again, not sure whether that would be correct. It would be best if we run this locally first with these changes.

---

Yes, it's tested by a trial release on the forked repo here: https://github.com/tony-josi-aws/FreeRTOS-Plus-TCP/actions/runs/12252149023

The `"0,/^ \* FreeRTOS\+TCP/` part of the script should apply the replacement only to the first occurrence of the specified pattern within the range from the start of the file (`0`) to the first matching line [`FreeRTOS\+TCP`].

---

This commit: https://github.com/tony-josi-aws/FreeRTOS-Plus-TCP/commit/e31765e3f2bec4cb295d21e25a99d299a19a0dfc
has the changes made by the script.

---

Makes sense! Thank you @tony-josi-aws.